libkmod/libkmod-index.c

```diff
@@ -451,7 +451,6 @@ static char *index_search__node(struct index_node_f *node, const char *key, int
  */
 char *index_search(struct index_file *in, const char *key)
 {
-	// FIXME: return value by reference instead of strdup
```

Comment:
I'd keep the comment. If @lucasdemarchi adds a strbuf backed by scratchbuf, then it might be possible to skip the strdup. After all the index_search function could trigger an allocation, while index_mm_search would initialize the scratch buffer with a pointer and proper size, so no free would be needed.

Haven't made any further progress in this, because it's one strdup per module insertion, too much exclusive scratchbuf-code to add for this and would duplicate future work.

Reply:
Sure, dropped.